bs/src/tablecloth.ml

```diff
@@ -6,6 +10,135 @@ let ( << ) (f1 : 'b -> 'c) (f2 : 'a -> 'b) : 'a -> 'c = fun x -> x |> f2 |> f1
 
 let identity (value : 'a) : 'a = value
 
+let flip f x y = f y x
+
+module Array = struct
+  let empty : 'a array = [||]
+
+  let singleton a = [|a|]
+
+  let length (a : 'a array) : int = Belt.Array.length a
+
+  let isEmpty (a : 'a array) : bool = length a = 0
+
+  let is_empty = isEmpty
+
+  let initialize ~length ~f = Belt.Array.makeBy length f
+
+  let repeat ~length (e: 'a): 'a array = Belt.Array.make length e
+
+  let range ?(from = 0) (to_ : int) : int array =
+    Belt.Array.makeBy (to_ - from) (fun i -> i + from)
+
+  let fromList (l: 'a list) : 'a array = Belt.List.toArray l
+
+  let from_list = fromList
+
+  let toList (a: 'a array) : 'a list = Belt.List.fromArray a
+
+  let to_list = toList
+
+  let toIndexedList array = 
+    Belt.Array.reduceReverse array (length array - 1, []) (fun (i, acc) x -> 
+      (i - 1, ((i, x) :: acc)))
+    |> snd
+
+  let to_indexed_list = toIndexedList
+
+  let get ~index array = Belt.Array.get array index 
+
+  let set ~index ~value array = array.(index) <- value
+
+  let sum (a : int array) : int = Belt.Array.reduce a 0 ( + )
+
+  let floatSum (a : float array) : float = Belt.Array.reduce a 0.0 ( +. )
+
+  let float_sum = floatSum
+
+  let filter ~(f : 'a -> bool) (a : 'a array) : 'a array = Belt.Array.keep a f
+
+  let map ~(f : 'a -> 'b) (a : 'a array) : 'b array = Belt.Array.map a f
+
+  let mapWithIndex ~(f : 'int -> 'a -> 'b) (a : 'a array) : 'b array = Belt.Array.mapWithIndex a f
+
+  let map_with_index = mapWithIndex
+
+  let mapi = mapWithIndex
+
+  let map2 ~(f : 'a -> 'b -> 'c) (a : 'a array) (b : 'b array) : 'c array = Belt.Array.zipBy a b f
+
+  let map3 ~(f : 'a -> 'b -> 'c -> 'd) (arrayA : 'a array) (arrayB : 'b array) (arrayC : 'c array) : 'd array =
+    let minLength : int = Belt.Array.reduce [|length arrayB; length arrayC|] (length arrayA) min in
+    Belt.Array.makeBy minLength (fun i -> f arrayA.(i) arrayB.(i) arrayC.(i))
+
+  let flatMap ~(f : 'a -> 'b array) (a : 'a array) : 'b array = Belt.Array.map a f |> Belt.Array.concatMany
+
+  let flat_map = flatMap
+
+  let find ~(f : 'a -> bool) (array : 'a array) : 'a option = 
+    let rec find_loop array ~f ~length i =
+      if i >= length then None
+      else if f array.(i) then Some (array.(i))
+      else find_loop array ~f ~length (i + 1)
+    in
+    find_loop array ~f ~length:(length array) 0 
+
+  let any ~(f : 'a -> bool) (a : 'a array) : bool = Belt.Array.some a f
+
+  let all ~(f : 'a -> bool) (a : 'a array) : bool = Belt.Array.every a f
+
+  let append (a : 'a array) (a' : 'a array)  : 'a array = Belt.Array.concat a a'
+
+  let concatenate (ars : 'a array array) : 'a array = Belt.Array.concatMany ars
+
+  let flatten = concatenate
+
+  let intersperse ~(sep : 'a) (array : 'a array) : 'a array = 
+    Array.init
```

Comment:
Array.init -> initialize

Reply:
This is currently using the standard libraries implementation of init

Reply:
you can use initialize which has been implemented further up, this way you get the Base/Belt version no matter which file it is